lib/AST/ModuleNameLookup.cpp

```diff
@@ -127,6 +127,18 @@ void ModuleNameLookup<LookupStrategy>::lookupInModule(
     const DeclContext *moduleScopeContext) {
   assert(moduleOrFile->isModuleScopeContext());
 
+  // Does the module scope have any separately-imported overlays shadowing
+  // the module we're looking into?
+  SmallVector<ModuleDecl *, 4> overlays;
+  moduleScopeContext->getSeparatelyImportedOverlays(
+      moduleOrFile->getParentModule(), overlays);
+  if (!overlays.empty()) {
+    // If so, look in each of those overlays.
+    for (auto overlay : overlays)
+      lookupInModule(decls, overlay, accessPath, moduleScopeContext);
```

Comment:
Will this work with multiple overlays on a single underlying module? Need a test for this.